src/typing/type_inference_js.ml

```diff
@@ -4644,15 +4644,14 @@ and mk_signature cx reason_c c_type_params_map superClass body = Ast.Class.(
         static;
       }) ->
 
-      let typeparams, f_type_params_map =
-        mk_type_param_declarations cx ~map:c_type_params_map typeParameters in
-
-      let map = SMap.fold SMap.add f_type_params_map c_type_params_map in
+      (* SG: Do we really need the separate type_params_map here? *)
+      let typeparams, type_params_map =
+        mk_type_param_declarations cx type_params_map typeParameters in
 
-      let params_ret = mk_params_ret cx map
+      let params_ret = mk_params_ret cx type_params_map
         (params, defaults, rest) (body_loc body, returnType) in
       let reason_m = mk_reason (spf "method %s" name) loc in
-      let method_sig = reason_m, typeparams, f_type_params_map, params_ret in
+      let method_sig = reason_m, typeparams, type_params_map, params_ret in
```

Comment:
Before, `method_sig` would only have a map with the type params in `typeparams`, but after this change it will have all the type params accumulated up to this point. Did I inadvertently break something with this change?